VMRecovery/Testing/Create-TestVM.ps1

```diff
@@ -0,0 +1,848 @@
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftWindowsServer' -Offer 'WindowsServer' -Skus '2016-Datacenter-smalldisk')[-1]
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftWindowsServer' -Offer 'WindowsServer' -Skus '2012-R2-Datacenter-smalldisk')[-1]
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftWindowsServer' -Offer 'WindowsServer' -Skus '2012-Datacenter-smalldisk')[-1]
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftWindowsServer' -Offer 'WindowsServer' -Skus '2008-R2-SP1-smalldisk')[-1]
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftWindowsDesktop' -Offer 'Windows-10' -Skus 'rs5-pro')[-1]
+        }
+
+        if ($createWin7andWin8)
+        {
+            $windowsImages = @()
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftVisualStudio' -Offer 'Windows' -Skus 'Win7-SP1-Ent-N-x64')[-1]
+            $windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftVisualStudio' -Offer 'Windows' -Skus 'Win81-Ent-N-x64')[-1]
+            #$windowsImages += (get-azurermvmimage -Location $location -PublisherName 'MicrosoftVisualStudio' -Offer 'Windows' -Skus 'Windows-10-N-x64')[-1]
+            if($windowsImages.Count -ne 2)
+            {
+                show-progress "No MicrosoftVisualStudio client images found. Make sure you are using an MSDN subscription"
+            }
+        }
+
+        # Add OSType because there are Windows vs. Linux specific inputs needed for provisioning
+        $windowsImages | foreach {$_ | Add-Member -MemberType NoteProperty -Name 'OsType' -Value 'Windows' -Force}
+    }
+
+    #Linux
+    if (!$windowsVMsOnly)
+    {
+        if (!$windows)
+        {
+            if ($publisherName -and $offer -and $skus)
+            {
+                if ($version)
+                {
+                    $linuxImages += get-azurermvmimage -Location $location -PublisherName $publisherName -Offer $offer -Skus $skus -version $version
+                }
+                else
+                {
+                    $linuxImages += (get-azurermvmimage -Location $location -PublisherName $publisherName -Offer $offer -Skus $skus)[-1]
+                }
+            }
+            else
+            {
+                $linuxImages += (get-azurermvmimage -Location $location -PublisherName 'Canonical' -Offer 'UbuntuServer' -Skus '18.04-LTS')[-1]
+            }
+        }
+
+        if (!$windows -and !$linux -and !$imageName)
+        {
+            $linuxImages += (get-azurermvmimage -Location $location -PublisherName 'CoreOS' -Offer 'CoreOS' -Skus 'Stable')[-1]
```

Comment:
Lots of distro specific strings, etc sprinkled throughout the scirpt that is subject to change.  Similar to the above you might consider reading this from a config file to make it easier to manage.  Otherwise when a newer version is available there will be lots of edits.

Reply:
Removed the testing folder